content/courses/intro-to-ada/chapters/fixed_point_types.rst

```diff
@@ -206,42 +259,81 @@ We may also use any other range. For example:
-       B := 0.50;
-       R := A + B;
-       Put_Line ("R is " & TQ31'Image (R));
-    end Fixed_Point_Op;
+The main difference between fixed-point and floating-point types is that
+fixed-point types don't have an exponent. This has an impact on calculations
+using small values: while they might still be representable with floating-point
+types, those small values might simply *disappear* (i.e. become zero) in the
+fixed-point representation. Let's see an example where we compare the decimal
+type :ada:`Decimal` to the floating-point type :ada:`Float_32`:
 
-As expected, :ada:`R` contains 0.75 after the addition of :ada:`A` and :ada:`B`.
+.. code:: ada run_button project=Courses.Intro_To_Ada.Fixed_Point_Types.Decimal_Vs_Floating_Point_Types
 
-In fact the language is more general than these examples imply, since in
-practice it is typical to need to multiply or divide values from different
-fixed-point types, and obtain a result that may be of a third fixed-point type.
-The details are outside the scope of this introductory course.
+    with Ada.Text_IO; use Ada.Text_IO;
 
-It is also worth noting, although again the details are outside the scope of
-this course, that you can explicitly specify a value for an ordinary
-fixed-point type's :ada:`small`.  This allows non-binary scaling, for example:
+    procedure Decimal_Vs_Floating_Point_Types is
+       type Decimal is
+         delta 10.0 ** (-2) digits 9;
 
-.. code-block:: ada
+       type Float_32 is
+         digits 6
+         range -9999999.99 .. 9999999.99;
 
-    type Angle is
-      delta 1.0/3600.0
-      range 0.0 .. 360.0 - 1.0 / 3600.0;
-    for Angle'Small use Angle'Delta;
+       D : Decimal  := 0.01;
+       F : Float_32 := 0.01;
+    begin
+       Put_Line ("D = " &
+                 D'Image);
+       Put_Line ("F = " &
+                 F'Image);
+
+       D := D / 2.0;
+       --   ^^^^^^^
+       --  Value becomes zero.
+
+       F := F / 2.0;
+       --   ^^^^^^^
+       --  Exponent is used to
+       --  represent smaller
+       --  value.
+
+       Put_Line ("D = " &
+                 D'Image);
+       Put_Line ("F = " &
+                 F'Image);
+    end Decimal_Vs_Floating_Point_Types;
+
+Both types in this example have roughly the same size and range. However, the
+result of the divide-by-two operation isn't the same: because of the exponent,
+:ada:`F` has the expected value (0.005) after the operation. while the value of
+:ada:`D` is zero. The reason is that the resulting value 0.005 cannot be
+represented by the decimal precision of the :ada:`Decimal` type. In the case of
+:ada:`F`, however, the value can be represented due to a simple change in the
+exponent.
+
+This lack of precision we just described might seem like a drawback for
+fixed-point types. However, depending on the algorithm and its field of
+application, this is the exact behavior that we might be looking for. As
+mentioned in the beginning of this chapter, financial applications might
```

Comment:
I suggest striking "might" since financial applications are the reason decimal fixed-point types were added to the language.

Reply:
Yes, you're right!